app/models/due_date.rb

```diff
@@ -83,4 +83,13 @@ def self.get_next_due_date(assignment_id, topic_id = nil)
     end
     next_due_date
   end
+  
+  #Find the due date at which the assignment was Finished.
+  def self.get_past_due_date(assignment_id, topic_id = nil)
```

Comment:
This method looks not correct. You are looking at the first `AssignmentDueDate` and assign it to `Finish`?